mergify_engine/json.py

```diff
@@ -0,0 +1,51 @@
+# -*- encoding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+import enum
+import json
+
+
+JSON_TYPES = {}
+
+
+def register_type(cls):
+    JSON_TYPES[cls.__name__] = cls
+
+
+class EnumEncoder(json.JSONEncoder):
+    def default(self, v):
+        if isinstance(v, enum.Enum):
+            return {
+                "__pytype__": "enum",
+                "class": type(v).__name__,
+                "name": v.name,
+            }
+        else:
+            return super().default(v)
+
+
+def decode_enum(dct):
+    if "enum" in dct.values():
```

Comment:
This is going to load ALL the values of the dict and check in ALL of them… There's a better way to do this!